### PR TITLE
kvserver: eagerly unquiesce ranges that use leader leases

### LINKED ISSUE #136231: kvserver: eagerly unquiesce ranges that use leader leases
**Describe the problem**

We eagerly unquiesce expiration based leases when starting/restarting a store. Like expiration based leases, leader leases don't support quiescence. As such, I don't think they shouldn't begin life as quiescent.

cc @nvanbenschoten 

Jira issue: CRDB-44934

### PR DESCRIPTION
**Pull request by arulajmani**
We eagerly unquiesce expiration based leases when starting/restarting a store. Like expiration based leases, leader leases also don't support quiescence. As such, this patch has them start off life as unquiescent.

Found while trying to port TestStoreLoadReplicaQuiescent over to use leader leases.

References https://github.com/cockroachdb/cockroach/issues/133763 
Closes https://github.com/cockroachdb/cockroach/issues/136231

Release note: None

### CODE REVIEW

**Review by iskettaneh — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136232)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arulajmani and @nvanbenschoten)

___
*[`pkg/kv/kvserver/client_store_test.go` line 169 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/136232#-OCnMN_nD0DUYCSo1UoM:-OCnMN_nD0DUYCSo1UoN:b3upiii) ([raw file](https://github.com/cockroachdb/cockroach/blob/1aea6d46e2ed7323c51508b93be783fd32368bbe/pkg/kv/kvserver/client_store_test.go#L169)):*
> ```Go
> 
> 		if leaseType == roachpb.LeaseLeader {
> 			// TODO(arul): add words.
> ```

Nit: Fix comment

File: pkg/roachpb/data.go [hunk 1]
```diff
@@ -2004,6 +2004,18 @@ func (l Lease) Type() LeaseType {
 	return LeaseExpiration
 }
 
+// SupportsQuiescence returns whether the lease supports quiescence or not.
+func (l Lease) SupportsQuiescence() bool {
```
Comment (nvanbenschoten) [inline pkg/roachpb/data.go:2008]: Do we want to call this from `shouldReplicaQuiesceRaftMuLockedReplicaMuLocked`?

**Review by arulajmani — COMMENTED** [review]
RFAL

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136232#-:-ODha3B_5dQwqmwDerzM:bb72pd0)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @iskettaneh and @nvanbenschoten)

**Review by nvb — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136232#-:-ODhfxJ_0qaBlekDWP3r:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @arulajmani and @iskettaneh)

___
*[`-- commits` line 12 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/136232#-ODhfMsD9BccxTML55m2:-ODhfMsD9BccxTML55m3:b-pgzavp):*
Mind adding that this fixed https://github.com/cockroachdb/cockroach/issues/135270 as well?
___
*[`pkg/kv/kvserver/store.go` line 2332 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/136232#-ODhfhbW7FbgyRCIt_5Z:-ODhfhbW7FbgyRCIt_5_:bqz9zav) ([raw file](https://github.com/cockroachdb/cockroach/blob/4e40d3c3abefbb72e237656b831c6c3b1cad3de5/pkg/kv/kvserver/store.go#L2332)):*
> ```Go
> 		// or expiration leases. We want to eagerly establish raft leaders for such
> 		// ranges and acquire leases on top of this. This happens during Raft ticks.
> 		// We rely on Raft pre-vote to avoid disturbance to Raft leader.s
> ```

"leader.s"

**Review by arulajmani — COMMENTED** [review]
Thanks!

bors r=nvanbenschoten

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136232#-:-ODhszvh1f7CQXUh9noV:bh7uere)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @iskettaneh and @nvanbenschoten)

___
*[`pkg/kv/kvserver/store.go` line 2332 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/136232#-ODhfhbW7FbgyRCIt_5Z:-ODhsn3aCmSl8JQ0Qx1D:beuwnk7) ([raw file](https://github.com/cockroachdb/cockroach/blob/4e40d3c3abefbb72e237656b831c6c3b1cad3de5/pkg/kv/kvserver/store.go#L2332)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

"leader.s"
</blockquote></details>

Hazards of typing too fast :sweat_smile:

**Comment by arulajmani** [comment]
bors retry

**Pull request merged**